zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -101,15 +107,101 @@ private[codec] object EncoderDecoder                   {
 
     private val flattened: AtomizedCodecs = AtomizedCodecs.flatten(httpCodec)
 
-    private val jsonEncoders = flattened.content.map { bodyCodec =>
+    private val jsonEncoders      = flattened.content.map { bodyCodec =>
       val erased    = bodyCodec.erase
       val jsonCodec = JsonCodec.schemaBasedBinaryCodec(erased.schema)
       erased.encodeToBody(_, jsonCodec)
     }
-    private val jsonDecoders = flattened.content.map { bodyCodec =>
+    private val formFieldEncoders = flattened.content.map { bodyCodec => (name: String, value: Any) =>
+      {
+        val erased = bodyCodec.erase
+        erased.schema match {
+          case Schema.Primitive(_, _) =>
+            FormField.simpleField(name, value.toString)
+          case _                      =>
+            val jsonCodec = JsonCodec.schemaBasedBinaryCodec(erased.schema)
+            FormField.textField(
+              name,
+              new String(jsonCodec.encode(value.asInstanceOf[erased.Element]).toArray, Charsets.Utf8),
+              MediaType.application.json,
+            )
+        }
+      }
+    }
+    private val jsonDecoders      = flattened.content.map { bodyCodec =>
       val jsonCodec = JsonCodec.schemaBasedBinaryCodec(bodyCodec.schema)
       bodyCodec.decodeFromBody(_, jsonCodec)
     }
+    private val formFieldDecoders = flattened.content.map { bodyCodec => (field: FormField) =>
+      {
+        def fieldAsText = field match {
+          case FormField.Text(_, value, _, _)                =>
+            ZIO.succeed(value)
```

Comment:
This seems incorrect to decode every text field into a string. What if it's an int or other `StandardType` that was converted into a `String`?